pysteps/utils/arrays.py

```diff
@@ -11,6 +11,81 @@
 """
 
 import numpy as np
+from numpy.lib.stride_tricks import as_strided
+
+
+def block_reduce(image, block_size, func=np.mean, **kwargs):
+    """Downsample image by applying a function to local blocks
+    (average by default).
```

Comment:
This new method seems very similar to the already existing `utils.dimensions.aggregate_fields_space` method, although with improved functionalities (e.g., data are trimmed if not perfectly divisible).

Do we need to keep both? Can we have only one method to upscale/reduce/aggregate fields in space?

Reply:
Oops, my bad. I totally forgot about the functions in the `utils.dimensions` module. Commit 7e493f6 refactors the 'aggregate_fields' function to include the functionality of the  `block_reduce` function (now removed).  Also, I added a few more tests to check the `aggregate_fields` function under different scenarios.

Reply:
Nice job!